src/core/Akka.MultiNodeTestRunner/Discovery.cs

```diff
@@ -92,10 +92,10 @@ private IEnumerable<RoleName> RoleNames(Type specType)
             var args = ConfigConstructorParamValues(configType);
             var configInstance = Activator.CreateInstance(configType, args);
             var roleType = typeof(RoleName);
-            var configProps = configType.GetProperties(BindingFlags.DeclaredOnly | BindingFlags.Instance | BindingFlags.Public);
-            var roleProps = configProps.Where(p => p.PropertyType == roleType).Select(p => (RoleName)p.GetValue(configInstance));
-            var configFields = configType.GetFields(BindingFlags.DeclaredOnly | BindingFlags.Instance | BindingFlags.Public);
-            var roleFields = configFields.Where(f => f.FieldType == roleType).Select(f => (RoleName)f.GetValue(configInstance));
+            var configProps = configType.GetProperties(BindingFlags.Instance | BindingFlags.Public);
+            var roleProps = configProps.Where(p => p.PropertyType == roleType && p.Name != "Myself").Select(p => (RoleName)p.GetValue(configInstance));
```

Comment:
@Horusiath errp.... Looks like these were debug lines added when you were testing this stuff. Going to need to remove that.

Reply:
No actually those changes are valid. Without them you cannot make inherited role names i.e. if you defined roles First, Second in some base `MultiNodeConfig` class, then inherit from it (like in the case of  `ClusterShardingSpecConfig`&rarr;`PersistentClusterShardingSpecConfig` and `ClusterShardingSpecConfig`&rarr;`DDataClusterShardingSpecConfig`), then MNTK runner won't detect those role names in actual specs, making the test cluster empty.

Reply:
What about the ` p.Name != "Myself"` - that looks like a one-off to me? No?

Reply:
By default, every `MultiNodeConfig` contains a property called Myself, which is also a `RoleName`. While we may want to spin cluster nodes based on properties like Controller, First, Second, Third etc., I guess we don't want to spin a cluster node called Myself ;) Especially since it has some specific meaning in context of a config.

I was thinking about excluding properties to be used as test cluster nodes by attaching some special attribute to them, but it seem to be too much ceremony - so far it's used only in one case.

Reply:
@Horusiath got it. Ok then, objection overruled. This is good to go.